widget/playground/src/components/TokensPanel/TokensPanel.tsx

```diff
@@ -26,14 +28,114 @@ export function TokensPanel(props: PropTypes) {
     list: listProps,
     selectedBlockchains: selectedBlockchainsProps,
     onChange,
+    configTokens,
   } = props;
   const [selectedBlockchain, setSelectedBlockchain] = useState(
     selectedBlockchainsProps[0]
   );
+
   const [list, setList] = useState(listProps);
+
+  const [supportedTokenList, setSupportedTokenList] = useState(
+    configTokens || {}
+  );
   const [showSelectedTokens, setShowSelectedTokens] = useState(false);
 
+  const getTokens = (
+    selected: boolean,
+    allTokensInBlockchain: Asset[],
+    token?: TokenType
+  ) => {
+    if (!token) {
+      return selected ? allTokensInBlockchain : [];
+    }
+    const { blockchain, symbol, address } = token;
+    if (!selected) {
+      return [
+        ...supportedTokenList[blockchain].tokens.filter(
+          (t) => !tokensAreEqual(t, token)
+        ),
+      ];
+    }
+    return [
+      ...supportedTokenList[blockchain].tokens,
+      { symbol, address, blockchain },
+    ];
+  };
+
+  /*
+   * This function works for three actions:
+   * . Click on the checkbox
+   * . Click Select All
+   * . Click on select all in each blockchain
+   * If the token has been sent, it means that the checkbox has been clicked
+   * And if it has not been sent, it means that one of the other two options has been clicked
+   * The output is what we need for configuration
+   */
+
+  const makeSupportedTokenList = (
+    blockchain: string,
+    tokens: {
+      [blockchain: string]: Tokens;
+    },
+    selected: boolean,
+    token?: TokenType
+  ) => {
+    const allTokensInBlockchain = list
+      .filter((item) => item.blockchain === blockchain)
+      .map(({ symbol, blockchain, address }) => ({
+        symbol,
+        blockchain,
+        address,
+      }));
+    if (supportedTokenList[blockchain]) {
+      const blockchainTokens = supportedTokenList[blockchain].tokens;
+      if (!supportedTokenList[blockchain].isExcluded) {
+        if (
+          (token &&
+            !token.checked &&
+            blockchainTokens.length + 1 === allTokensInBlockchain.length) ||
+          (!token && selected)
+        ) {
+          const { [blockchain]: deletedKey, ...otherKeys } = tokens;
+          return otherKeys;
+        }
+      }
+
+      return {
+        ...tokens,
+        [blockchain]: {
+          ...tokens[blockchain],
+          tokens: getTokens(selected, allTokensInBlockchain, token),
+        },
+      };
+    }
+    if (!token && selected) {
+      return {};
+    }
+    return {
+      ...tokens,
+      [blockchain]: {
+        isExcluded: false,
+        tokens:
+          !token && !selected
+            ? []
+            : allTokensInBlockchain.filter((t) => !tokensAreEqual(t, token)),
+      },
+    };
+  };
```

Comment:
When user deselect all tokens, we add a lot of necessary data to the exported config. We need a solution for this.

```
const config = {
  from: {
      pinnedTokens: [],
      tokens: {
          ETH: {
              isExcluded: false,
              tokens: []
          },
          BSC: {
              isExcluded: false,
              tokens: [
                  {
                      symbol: 'ETH',
                      address: '0x2170ed0880ac9a755fd29b2688956bd959f933f8',
                      blockchain: 'BSC'
                  },
                  {
                      symbol: 'USDT',
                      address: '0x55d398326f99059ff775485246999027b3197955',
                      blockchain: 'BSC'
                  }
              ]
          },
          ARBITRUM: {
              isExcluded: true,
              tokens: [
                  {
                      symbol: 'ETH',
                      address: null,
                      blockchain: 'ARBITRUM'
                  }
              ]
          },
          POLYGON: {
              isExcluded: false,
              tokens: []
          },
          ZKSYNC: {
              isExcluded: false,
              tokens: []
          },
          STARKNET: {
              isExcluded: false,
              tokens: []
          },
          OPTIMISM: {
              isExcluded: false,
              tokens: []
          },
          AVAX_CCHAIN: {
              isExcluded: false,
              tokens: []
          },
          POLYGONZK: {
              isExcluded: false,
              tokens: []
          },
          BASE: {
              isExcluded: false,
              tokens: []
          },
          LINEA: {
              isExcluded: false,
              tokens: []
          },
          TRON: {
              isExcluded: false,
              tokens: []
          },
          BTC: {
              isExcluded: false,
              tokens: []
          },
          COSMOS: {
              isExcluded: false,
              tokens: []
          },
          OSMOSIS: {
              isExcluded: false,
              tokens: []
          },
          METIS: {
              isExcluded: false,
              tokens: []
          },
          NEUTRON: {
              isExcluded: false,
              tokens: []
          },
          NOBLE: {
              isExcluded: false,
              tokens: []
          },
          DYDX: {
              isExcluded: false,
              tokens: []
          },
          SOLANA: {
              isExcluded: false,
              tokens: []
          },
          CRONOS: {
              isExcluded: false,
              tokens: []
          },
          BNB: {
              isExcluded: false,
              tokens: []
          },
          AURORA: {
              isExcluded: false,
              tokens: []
          },
          MAYA: {
              isExcluded: false,
              tokens: []
          },
          THOR: {
              isExcluded: false,
              tokens: []
          },
          BOBA: {
              isExcluded: false,
              tokens: []
          },
          MOONBEAM: {
              isExcluded: false,
              tokens: []
          },
          MOONRIVER: {
              isExcluded: false,
              tokens: []
          },
          OKC: {
              isExcluded: false,
              tokens: []
          },
          LTC: {
              isExcluded: false,
              tokens: []
          },
          BCH: {
              isExcluded: false,
              tokens: []
          },
          HECO: {
              isExcluded: false,
              tokens: []
          },
          STARGAZE: {
              isExcluded: false,
              tokens: []
          },
          CRYPTO_ORG: {
              isExcluded: false,
              tokens: []
          },
          CHIHUAHUA: {
              isExcluded: false,
              tokens: []
          },
          BANDCHAIN: {
              isExcluded: false,
              tokens: []
          },
          COMDEX: {
              isExcluded: false,
              tokens: []
          },
          REGEN: {
              isExcluded: false,
              tokens: []
          },
          IRIS: {
              isExcluded: false,
              tokens: []
          },
          EMONEY: {
              isExcluded: false,
              tokens: []
          },
          JUNO: {
              isExcluded: false,
              tokens: []
          },
          STRIDE: {
              isExcluded: false,
              tokens: []
          },
          MARS: {
              isExcluded: false,
              tokens: []
          },
          TERRA: {
              isExcluded: false,
              tokens: []
          },
          BITSONG: {
              isExcluded: false,
              tokens: []
          },
          AKASH: {
              isExcluded: false,
              tokens: []
          },
          KI: {
              isExcluded: false,
              tokens: []
          },
          PERSISTENCE: {
              isExcluded: false,
              tokens: []
          },
          MEDIBLOC: {
              isExcluded: false,
              tokens: []
          },
          KUJIRA: {
              isExcluded: false,
              tokens: []
          },
          SENTINEL: {
              isExcluded: false,
              tokens: []
          },
          INJECTIVE: {
              isExcluded: false,
              tokens: []
          },
          SECRET: {
              isExcluded: false,
              tokens: []
          },
          KONSTELLATION: {
              isExcluded: false,
              tokens: []
          },
          STARNAME: {
              isExcluded: false,
              tokens: []
          },
          BITCANNA: {
              isExcluded: false,
              tokens: []
          },
          UMEE: {
              isExcluded: false,
              tokens: []
          },
          DESMOS: {
              isExcluded: false,
              tokens: []
          },
          LUMNETWORK: {
              isExcluded: false,
              tokens: []
          }
      }
  },
}
```

Reply:
This is the expected config in my opinion in this case:
So I think when user deselect all blockchains tokens in tokens section in playground, one possible solution is to update blockchains filter and remove those blockchains that user deselect all tokens for them. 

```
blockchains: [only blockchains that have some tokens],
tokens: {
  BSC: {
      isExcluded: false,
      tokens: [
          {
              symbol: 'ETH',
              address: '0x2170ed0880ac9a755fd29b2688956bd959f933f8',
              blockchain: 'BSC'
          },
          {
              symbol: 'USDT',
              address: '0x55d398326f99059ff775485246999027b3197955',
              blockchain: 'BSC'
          }
      ]
  },
  ARBITRUM: {
      isExcluded: true,
      tokens: [
          {
              symbol: 'ETH',
              address: null,
              blockchain: 'ARBITRUM'
          }
      ]
  },
 }
 ```

Reply:
As we talked, we need these changes:

- [ ] change Select All Tokens to Reset
- [ ] change label Included/Excluded based on check box

Reply:
fixed